tests/MongoDB.Driver.Tests/Specifications/unified-test-format/UnifiedTestOperations/IUnifiedTestOperation.cs

```diff
@@ -0,0 +1,26 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace MongoDB.Driver.Tests.Specifications.unified_test_format.UnifiedTestOperations
+{
+    public interface IUnifiedTestOperation
```

Comment:
Justification for introducing new way of creating and representing operation types:
1. In new spec operation results and exceptions are unified and asserted with common logic by converting natural operation result into BsonDocument and validating according to [Evaluating Matches](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#evaluating-matches) which removes the need for methods `Assert`, `AssertResult`, `AssertException`, `ParseExpectedResult` from `JsonDrivenTest`.
2. Existing `JsonDrivenTest` has a complex hierarchy (abstract classes: `JsonDrivenCommandTest`, `JsonDrivenCollectionTest`, `JsonDrivenSessionTest`, `JsonDrivenTestRunnerTest`, `JsonDrivenWithThreadTest`) which makes it hard to navigate through existing json test classes and their logic.
3. Semantics and responsibility narrowing: while `JsonDrivenTest` represents an operation which does arrangement, execution and assertion, `IUnifiedTestOperation` describes an operation which is only executed, which better compels to the unified test spec operation concept ([operation](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#operation)).

Reply:
I think this interface and these new classes are clean and simple. Well done.